packages/client/src/views/VerifyCertificate/VerifyCertificatePage.tsx

```diff
@@ -0,0 +1,551 @@
+            city
+            districtName
+            stateName
+          }
+        }
+        registration {
+          trackingId
+          registrationNumber
+          type
+        }
+        createdAt
+        history {
+          action
+          regStatus
+          user {
+            name {
+              firstNames
+              familyName
+            }
+            catchmentArea {
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export function VerifyCertificatePage() {
+  const intl = useIntl()
+  const { declarationId } = useParams<{ declarationId: string }>()
+
+  const logo = useSelector(selectCountryLogo)
+  const appName = useSelector(selectApplicationName)
+
+  const [closeWindow, setCloseWindow] = React.useState(false)
+  const [timeOut, setTimeOut] = React.useState(false)
+
+  const { loading, error, data } = useQuery(
+    FETCH_RECORD_DETAILS_FOR_VERIFICATION,
+    {
+      variables: { id: declarationId },
+      fetchPolicy: 'network-only'
+    }
+  )
+
+  const [currentData, setCurrentData] = React.useState<
+    BirthRegistration | DeathRegistration
+  >()
+
+  React.useEffect(() => {
+    if (data) {
+      setTimeout(() => {
+        setCloseWindow(true)
+      }, 60000) // 60000 is correct value
+
+      setTimeout(() => {
+        setCloseWindow(false)
+        setTimeOut(true)
+      }, 600000) // 600000 is correct value
+      setCurrentData(data.fetchRecordDetailsForVerification)
+    }
+  }, [data])
+
+  const closeWindowAction = () => {
+    const blank = window.open('about:blank', '_self')
+    // @ts-ignore
+    blank.close()
+  }
+
+  const LoadingState = () => {
+    return (
+      <CheckingContainer>
+        <Stack alignItems="center" direction="column">
+          <SpaceDiv />
+          <SpinnerContainer>
+            <StyledSpinner id="appSpinner" />
+          </SpinnerContainer>
+          <Text variant={'reg16'} element={'h4'}>
+            {intl.formatMessage(messageToDefine.loadingState)}
+          </Text>
+        </Stack>
+      </CheckingContainer>
+    )
+  }
+
+  const ErrorState = () => {
+    return (
+      <>
+        <Alert onActionClick={() => {}} type="error">
+          <Text variant={'bold16'} element={'span'} color={'redDark'}>
+            {intl.formatMessage(messageToDefine.errorAlertTitle)}
+          </Text>{' '}
+          <br />
+          <Text variant={'reg16'} element={'span'} color={'redDark'}>
+            {intl.formatMessage(messageToDefine.errorAlertMessage)}
+          </Text>
+        </Alert>
+      </>
+    )
+  }
+
+  const TimeOutState = () => {
+    return (
+      <>
+        <CheckingContainer>
+          <Stack alignItems="center" direction="column">
+            <SpaceDiv />
+            <Text variant={'reg16'} element={'h4'}>
+              {intl.formatMessage(messageToDefine.timeOutState)}
+            </Text>
+            <SpaceDiv />
+          </Stack>
+        </CheckingContainer>
+      </>
+    )
+  }
```

Comment:
These can be moved into their own separate components. We should avoid writing a component inside another component